sycl/source/context.cpp

```diff
@@ -37,13 +37,26 @@ context::context(const vector_class<device> &DeviceList,
   if (DeviceList.empty()) {
     throw invalid_parameter_error("DeviceList is empty.");
   }
-  if (DeviceList[0].is_host())
-    impl = std::make_shared<detail::context_impl>(DeviceList[0], AsyncHandler);
-  else
-    // TODO also check that devices belongs to the same platform
-    impl = std::make_shared<detail::context_impl>(DeviceList, AsyncHandler);
+  auto number = std::find_if_not(
+      DeviceList.begin(), DeviceList.end(),
+      [&](const device &current_device) { return current_device.is_host(); });
+  if ( number == DeviceList.end() && DeviceList.back().is_host())
```

Comment:
Have you applied clang-format?

```suggestion
  if (number == DeviceList.end() && DeviceList.back().is_host())
```